tests/encode.rs

```diff
@@ -65,16 +65,17 @@ async fn encode_retry() -> http_types::Result<()> {
 #[async_std::test]
 async fn dropping_encoder() -> http_types::Result<()> {
     let (sender, encoder) = encode();
-    let reader = BufReader::new(encoder);
     let sender_clone = sender.clone();
-    task::spawn(async move { sender_clone.send("cat", "chashu", Some("0")).await.unwrap() });
+    task::spawn(async move { sender_clone.send("cat", "chashu", None).await });
 
-    //move the encoder into Lines, which gets dropped after this
-    assert_eq!(reader.lines().next().await.unwrap().unwrap(), "event:cat");
+    let mut reader = decode(BufReader::new(encoder));
+    let event = reader.next().await.unwrap()?;
+    assert_message(&event, "cat", "chashu", None);
+
+    std::mem::drop(reader);
 
-    assert_eq!(
-        sender.send("cat", "nori", None).await,
-        Err(async_sse::DisconnectedError)
-    );
+    assert!(matches!(sender.send("cat", "chashu", None).await,
+                     Err(e @ async_std::io::Error { .. })
+                            if e.kind() == async_std::io::ErrorKind::ConnectionAborted));
```

Comment:
This assert is a bit hard to follow.

Reply:
Agreed, will clean it up

Reply:
addressed in 95ddc4bf6862dec5f597d487104bf7826f79836d